### PR TITLE
[Feature Fix] Add user warning if navigate away after adding files for upload

### PR DESCRIPTION
**Pull request by geeksnglitter**
Purpose
-----------

When a user is on the dashboard and they decide to upload a file, it is easy to believe just dragging the file has uploaded it. In reality, the user must select a project and hit upload. If a user has started this process and tries to navigate away from the dashboard before completing it, a popup had been added to warn them that their changes will not be saved and gives them an option to stay on the page and finish.

Closes Issue CenterForOpenScience/openscienceframework.org#1518
Changes
------------

Adds a popup if the user tries to navigate away from the dashboard with an unfinished upload.

Side Effects
----------------

None

### CODE REVIEW

**Comment by samanehsan** [comment]
When a drag in a file, select a project and hit upload I get the pop-up telling me that I will loose these changes if I leave the page: 
screen shot 2015-06-03 at 3 31 46 pm

But the file finished uploading and the osf is trying to redirect me to the project page. Can you look into this?

**Comment by samanehsan** [comment]
Looks good! @sloria, I think this is good to go! :rainbow:

File: website/static/js/onboarder.js [hunk 1]
```diff
@@ -422,6 +422,8 @@ function OBUploaderViewModel(params) {
     self.messageClass = ko.observable('text-info');
     // The target node to upload to to
     self.target = ko.observable(null);
+    //Boolean to track of if upload was successful
```
Comment (samanehsan) [inline website/static/js/onboarder.js]: Oh wait, just some small language changes. Can you remove the extra `to` and the `if` in the comments?

**Comment by samanehsan** [comment]
Thanks! Now I think this is good to go :smile_cat:

File: website/static/js/onboarder.js [hunk 2]
```diff
@@ -576,6 +579,14 @@ function OBUploaderViewModel(params) {
         }
     };
     self.dropzone = new Dropzone(self.selector, dropzoneOpts);
+
+        //stop user from leaving if file is staged for upload
+        $(window).on('beforeunload', function() {
```
Comment (sloria) [inline website/static/js/onboarder.js]: Overindented. Will fix this during merge.

**Comment by sloria** [comment]
This branch is labeled as a hotfix, but I'm going to merge it as a feature into develop, since it isn't a critical fix. 

Good to merge. :octopus:

**Pull request merged**